ExpressWash/ExpressWash/CarController.swift

```diff
@@ -60,16 +60,44 @@ class CarController {
 
             guard let car = car else { return }
 
-            completion(car, nil)
-
             context.perform {
                 do {
                     try CoreDataStack.shared.save(context: context)
                 } catch {
                     print("Unable to update car: \(error)")
                     context.reset()
+                    completion(nil, error)
                 }
+                completion(car, nil)
+            }
+        }
+    }
+
+    func updateCarInCoreData(_ car: Car,
+                             rep: CarRepresentation,
+                             context: NSManagedObjectContext = CoreDataStack.shared.mainContext,
+                             completion: @escaping CompletionHandler) {
+        let privateMOC = NSManagedObjectContext(concurrencyType: .privateQueueConcurrencyType)
+        privateMOC.parent = context
+        privateMOC.performAndWait {
+            car.carId = Int32(rep.carId ?? NOID)
+            car.category = rep.category
+            car.clientId = Int16(rep.clientId)
+            car.color = rep.color
+            car.licensePlate = rep.licensePlate
+            car.make = rep.make
+            car.model = rep.model
+            car.photo = rep.photo
+            car.size = rep.size
+            car.year = rep.year
+            do {
+                try privateMOC.save()
+            } catch {
+                print("Unable to save updated car: \(error)")
+                context.reset()
+                completion(nil, error)
             }
+            completion(car, nil)
         }
```

Comment:
This function can go as discussed

Reply:
Changes didn't work correctly. Leaving as-is for now. Everything is working and we can change if needed.